pkg/ebpf/controlplane/controller.go

```diff
@@ -24,103 +26,155 @@ const (
 	cgroupRmdirControlProbe
 )
 
+// These events are needed for some of the control plane events. At the eBPF side, they also submit
+// events in the control plane perfbuffer. There isn't a need to set EventState for them (to
+// configure Submit to a policy number), but there IS a need for dealing with their dependencies:
+// making sure their probes are attached, make sure their dependencies are satisfied, make sure
+// their tailCalls are indexed, etc.
+var CoreEventsNeeded = []events.ID{
```

Comment:
See above, now we've coupled the control plane to events, which was undesirable in my opinion (if there must be coupling it would ideally be in the opposite direction IMO).

Reply:
Yep I had doubts about this as well. I started by having a required probed instead of events. But then there were the event dependencies and tailcall index enabling need, and things got out of control.

Reply:
Understandable, relates to program separation so see my comment above.

Reply:
Alright.